python/src/aiconfig/editor/client/src/components/aiconfigReducer.ts

```diff
@@ -181,22 +181,38 @@ function reduceConsolidateAIConfig(
       );
     }
     case "RUN_PROMPT": {
-      return reduceReplacePrompt(state, action.id, (prompt) => {
-        const responsePrompt = responseConfig.prompts.find(
-          (resPrompt) => resPrompt.name === prompt.name
-        );
+      // Note: We are relying on the callsite to explicitly set `isRunning`
+      // to true when it's needed. If isRunning is not explicitly defined
+      // in the action, and we have called consolidate aiconfig, then we
+      // assume that we are finished running the prompt.
```

Comment:
Side note, I left a comment on the PR that added this to the action but we should revisit it. Actions generally should not need to be passing state to set. We should determine if a prompt is running based on the action itself (i.e. maybe we add a different action RUN_PROMPT_STREAM_CHUNK to dispatch when a chunk is returned instead of using CONSOLIDATE_AICONFIG with RUN_PROMPT action).

We can revisit after gradio integration is complete, though

Reply:
> Actions generally should not need to be passing state to set.

Is this a general best practice, or more of your personalized opinion? 

I've also seen cases where the dispatch event types are more granular and therefore the event data passed along with it doesn't need to be as detailed. Ex:

```
dispatch.("RUN_PROMPT_START", {id: prompt.id})
...
dispatch("RUN_PROMPT_FINISH", {id: prompt.id})
```

^Do you have any preference?

Reply:
It's general best practice, just separates the concerns. Generally actions shouldn't need to worry about the underlying state, that logic should live within the reducer / state management.

More granular is fine as well; we could do the same and even pull the consolidate stuff out to explicit actions if desired. But not something to do anytime soon